cmd/utils/flags.go

```diff
@@ -805,8 +820,11 @@ func setWS(ctx *cli.Context, cfg *node.Config) {
 		}
 	}
 
-	if ctx.GlobalIsSet(WSPortFlag.Name) {
+	switch {
+	case ctx.GlobalIsSet(WSPortFlag.Name):
 		cfg.WSPort = ctx.GlobalInt(WSPortFlag.Name)
+	case ctx.GlobalBool(MixFlag.Name):
+		cfg.WSPort = 8646
```

Comment:
WS port should not be hardcoded. This is user-configurable, the user can choose any port arbitrarily, and it will continue to work properly with MIX Blockchain